### DISCUSSION TITLE
Custom Link grouped weights not working as expected

### DISCUSSION THREAD

**Discussion by AliMickey** [netbox-community/netbox#16234] [Help Wanted!]
Hi!

I have 4 custom links, 3 of which are in a group. The combined weight of the group is less than the 4th link however the group dropdown is marked as being 'heavier' and thus appears on the right side of the singular link.

![image](https://github.com/netbox-community/netbox/assets/60691199/7c1519bf-c692-427a-b5a2-ef8ccea0b062)

![image](https://github.com/netbox-community/netbox/assets/60691199/5930bead-7bb6-4fda-b1cd-583d8aa3f962)

Wasn't able to find any other mention of this, so not sure if bug or intended. 

Thanks.

**Comment by candlerb** [comment]
All ungrouped links are displayed first, followed by the groups. Source code: `netbox/extras/templatetags/custom_links.py`

The sorting appears to be implicit. From `netbox/extras/models/models.py`:

```
class CustomLink(CloningMixin, ExportTemplatesMixin, ChangeLoggedModel):
    ...
    class Meta:
        ordering = ['group_name', 'weight', 'name']
```

From this, I expect that if you have multiple groups, they ought to appear in a fixed alphabetical order. The "weight" only applies to the order of ungrouped links, and of links *within* a particular group.

**Comment by AliMickey** [comment]
Makes sense based on the behaviour.

I would suggest either to have separate group weights or count the total within a group (still allows for order within a group: 25,50,75 etc).
